### PR TITLE
Issue115 zero length file results in exception

### LINKED ISSUE #115: Zero Length VisualState.xml file results in exception
To reproduce:

1. Create a dll with tests or use an existing one, such as mock-assembly.dll

2. Create an empty visual state file of the same name, like mock-assembly.VisualState.xml

3. Try to open the file in the GUI.

Expected: GUI should simply ignore the empty file, just as it does for a non-existent file.

For other errors in the XML, the error should be reported but the GUI should not crash.

### PR DESCRIPTION
**Pull request by zBrianW**
Fixes #115 

Edit: Fixed comment to "auto close" issue

### CODE REVIEW

File: src/TestCentric/testcentric.gui/Views/TestTree.cs [hunk 1]
```diff
@@ -82,6 +82,10 @@ public string[] SelectedCategories
 
         private ITestModel Model { get; set; }
 
+        private readonly IMessageDisplay _messageDisplay;
+
+        private FileInfo _fileInfo;
```
[the remark this answers is not in the document]
Comment (zBrianW) [reply src/TestCentric/testcentric.gui/Views/TestTree.cs]: Removed `private FileInfo _fileInfo;`

File: src/TestCentric/testcentric.gui/Views/TestTree.cs [hunk 2]
```diff
@@ -82,6 +82,10 @@ public string[] SelectedCategories
 
         private ITestModel Model { get; set; }
 
+        private readonly IMessageDisplay _messageDisplay;
```
[the remark this answers is not in the document]
Comment (zBrianW) [reply src/TestCentric/testcentric.gui/Views/TestTree.cs]: Moved the newing of the MessageDisplay class into the catch.

File: src/TestCentric/testcentric.gui/Views/TestTree.cs [hunk 3]
```diff
@@ -430,17 +436,25 @@ public void InitializeView(ITestModel model)
...
-						model.SelectCategories(visualState.SelectedCategories, visualState.ExcludeCategories);
+					    try
+					    {
+					        VisualState visualState = VisualState.LoadFrom(fileName);
+					        tests.RestoreVisualState(visualState);
+					        model.SelectCategories(visualState.SelectedCategories, visualState.ExcludeCategories);
+                        }
+					    catch (Exception exception)
+					    {
+					        Console.WriteLine(exception);
```
[the remark this answers is not in the document]
Comment (zBrianW) [reply src/TestCentric/testcentric.gui/Views/TestTree.cs]: Removed the left behind `Console.WriteLine(exception);` line.

File: src/TestCentric/testcentric.gui/Views/TestTree.cs [hunk 4]
```diff
@@ -430,17 +436,25 @@ public void InitializeView(ITestModel model)
...
 					{
-						var visualState = VisualState.LoadFrom(fileName);
-						tests.RestoreVisualState(visualState);
-						model.SelectCategories(visualState.SelectedCategories, visualState.ExcludeCategories);
+					    try
+					    {
+					        VisualState visualState = VisualState.LoadFrom(fileName);
+					        tests.RestoreVisualState(visualState);
+					        model.SelectCategories(visualState.SelectedCategories, visualState.ExcludeCategories);
+                        }
```
[the remark this answers is not in the document]
Comment (zBrianW) [reply src/TestCentric/testcentric.gui/Views/TestTree.cs]: Formatted document (to replace tabs to spaces).

**Comment by zBrianW** [comment]
PR review updates.

**Comment by CharliePoole** [comment]
Once this is ready to merge, let's hold a beat. I gave a large set of diffs on play and I'm pretty sure we would lose this change if we let GitHub merge automatically since TestTree.cs disappears in my branch. Either @zBrianW or one of us can do a manual rebase and merge.

**Comment by CharliePoole** [comment]
Aside from the formatting diffs, this looks great to me. Thanks for doing it!

**Comment by CharliePoole** [comment]
@zBrianW Your changes can't be effectively rebased onto or merged with the master at this point, because the main file you changed, TestTree.cs, doesn't exist any longer. However, the code you changed does exist in TreeViewPresenter.cs, starting at line 70.

Would you mind re-applying the same key changes in that file, making a new PR?

**Comment by CharliePoole** [comment]
Yes, you can close as soon as you no longer want to be able to look at the diff.

**Comment by zBrianW** [comment]
Closing this PR - it has been replaced by https://github.com/TestCentric/testcentric-gui/pull/118 .

**Pull request closed without merging**